hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/OutboxImpl.java

```diff
@@ -90,11 +90,13 @@ public final boolean offer(int ordinal, @Nonnull Object item) {
 
     @Override
     public final boolean offer(int[] ordinals, @Nonnull Object item) {
-        if (numRemainingInBatch == 0) {
+        assert numRemainingInBatch != -1 : "Outbox.offer() called again after it returned false, without a " +
+                "call to resetBatch(). You probably didn't return from Processor method after Outbox.offer() " +
+                "or AbstractProcessor.tryEmit() returned false";
+        numRemainingInBatch--;
+        if (numRemainingInBatch == -1) {
```

Comment:
how about `if (--numRemainingInBatch == 0)` ? and replacing `-1`s with `0`s?

Reply:
it would be `if (numRemainingInBatch-- == 0)`, which is less readable imo, the `--` would be quite hidden.